src/openrct2/Game.cpp

```diff
@@ -1341,7 +1339,8 @@ void game_autosave()
         timeName, sizeof(timeName), "autosave_%04u-%02u-%02u_%02u-%02u-%02u%s", currentDate.year, currentDate.month,
         currentDate.day, currentTime.hour, currentTime.minute, currentTime.second, fileExtension);
 
-    limit_autosave_count(NUMBER_OF_AUTOSAVES_TO_KEEP, (gScreenFlags & SCREEN_FLAGS_EDITOR));
+    int32_t autosave_amount = gConfigGeneral.autosave_amount;
+    limit_autosave_count(autosave_amount-1, (gScreenFlags & SCREEN_FLAGS_EDITOR));
```

Comment:
Why `autosave_amount - 1`?

Reply:
The old code was not correct, this function makes sure there are X autosaves and removes older ones. The value of `NUMBER_OF_AUTOSAVES_TO_KEEP` was 9, but it kept 10 saves, because later in this function a new one will be made. I think this code is nicer.

I'd still request this variable to be renamed to something like `autosavesToKeep`. (And add spaces around the `-`).

Reply:
I haven't looked into the limit_autosave_count() but I tested it and autosave_amount +1 autosaves were saved without subtracting 1.